src/Raven.Studio/typescript/viewmodels/database/status/ongoingTasksStats.ts

```diff
@@ -1249,6 +1597,39 @@ class ongoingTasksStats extends viewModelBase {
             currentX += dx;
         }
     }
+
+    private drawSubscriptionStripes(context: CanvasRenderingContext2D, op: taskOperation,
+                                    xStart: number, yStart: number, yOffset: number, extentFunc: (duration: number) => number,
+                                    perfItemWithCache: SubscriptionConnectionPerformanceStatsWithCache, trackName: string) {
+
+        const dx = extentFunc(op.DurationInMs);
+
+        // Draw item
+        context.fillStyle = this.getColorForOperation(op.Name);
+        context.fillRect(xStart, yStart, dx, ongoingTasksStats.trackHeight);
+
+        // Register items
+        const itemToRegister = {
+            title: "Client Connection",
+            batchCount: perfItemWithCache.BatchCount,
+            connectionId: perfItemWithCache.ConnectionId,
+            duration: perfItemWithCache.Details.Operations[1].DurationInMs,
```

Comment:
is Operations[1] always available? What if operation is in progress?

Reply:
I check this condition:
`if (perfWithCache.Details.Operations.length > 1) {`
**before** i call `drawSubscriptionStripes`  (which registers the items..)
so it works just fine